### DISCUSSION TITLE
Download table which includes icons

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1624] [Q&A] [answered]
Hi team,

I was trying to download the content in my table as a CSV. In table it contains icons in some of the columns as below. 
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/101506482/190573473-c66ea98f-0a2d-4f08-bed7-ba7b72c7ac8b.png">
 The default download feature in the table downloads the content as a CSV. But icons are missing in the downloaded file. 
 
 Other than adding separate button/link to download the file is it possible to get download this file with raw content from default download feature in wave tables (Even in Excel file format)?

**Comment by aalencar** [comment]
Hi @shamilh2o,
What do you mean by raw content?

**Reply by shamilh2o** [reply]
Hi @aalencar 
I mean the file containing those icons as it is.

Expected:
![image](https://user-images.githubusercontent.com/101506482/190583772-1a99720a-b176-4751-a361-f859f071ffbf.png)

But those icons shown as below in downloaded file.
![image](https://user-images.githubusercontent.com/101506482/190583947-5202374c-d6b2-4a75-9e79-f6d9f7ea759a.png)

**Reply by aalencar** [reply]
Thanks for clarifying.

I have to check, but I believe all Wave icons are SVGs and those used by the OS are unicode-based. That would require us to manually create a map SVG -> unicode for each icon which is definitely not the way to go.

What if we exported the name of the icon and you replace them with the icon of your choice? Would that work for you? 

cc: @mturoci

**Reply by shamilh2o** [reply]
Can you please elaborate it @aalencar?

Currently I am assigning these icons ('❌' and '✅') as strings in the table row cells. 
If so is there a way to export a name instead of these icons with the wave table download option?

**Reply by aalencar** [reply]
Yes, that's what I meant. If you specify "check_mark" (I don't know if that's the proper icon name) it's gonna be displayed as ✅ in the table but when exported you will get "check_mark" in your column.

**Reply by shamilh2o** [reply]
Thanks for clarifying. 
In that case I believe we need to manually do some changes to the downloaded file. 
When a user accessing this feature I hope that will make trouble for them. I am expecting some approach that friendly for users. 

Anyway thanks for good suggestion.

**Comment by mturoci** [comment, marked as answer]
Seems like a relatively [easy feature](https://stackoverflow.com/questions/18249290/generate-csv-for-excel-via-javascript-with-unicode-characters). @shamilh2o can you please open a feature request? Thanks!

**Reply by shamilh2o** [reply]
Thanks @mturoci. 
Done https://github.com/h2oai/wave/issues/1626
